addons/block_code/blocks/communication/define_method.tres

```diff
@@ -5,11 +5,12 @@
 [resource]
 script = ExtResource("1_6e473")
 name = &"define_method"
+target_node_class = ""
 description = "Define a method/function with following statements"
 category = "Communication | Methods"
 type = 1
 variant_type = 0
-display_template = "Define method {method_name: NIL}"
+display_template = "Define method {method_name: STRING_NAME}"
```

Comment:
I also tried `TYPE_CALLABLE` since that's closer to what we're actually doing in this case. You can use a dummy callable like `Callable.create(null, 'my_method')` and later get the name out with `callable.get_method()`. Unfortunately, godot only supports serializing an empty `Callable()`, so the name would be lost in the serialization.

I tried hard to think of different alternatives, but I couldn't think of anything if we are completely in the world that inputs are encoded as `Variant` and only that is used to determine the type.

Reply:
I haven't read through the commits here so sorry if I'm missing details, but I wonder if the thing we need in the end here is really more in the code template end of things? We need a way to say "use this parameter as written, but don't quote it or coerce it into anything new". Could communicate that like `{{method_name}}` instead of `{method_name}`.

I didn't touch that, but I _was_ poking around here in #206 by removing the pretend `OPTION` parameter type and (as a result) using NIL as a type wherever we had used `OPTION`.

Reply:
The problem isn't so much in the template side. We can put whatever we want there and add something special. The problem comes in the generation side because everything is introspected based on the variant type. Even right now where this all happens in `ParameterInput.get_string`, everything happens based on `variant_type` except for the oddball case for `OPTION`. Actually, that part [looks pretty much the same](https://github.com/endlessm/godot-block-coding/blob/d8dae7e2f1f33248f5c039f2d03e401ae06c43bd/addons/block_code/code_generation/block_ast.gd#L139). The only reason `define_method` worked before this is that `EntryBlock` is treated specially, and [`get_entry_statement`](https://github.com/endlessm/godot-block-coding/blob/v0.6.2/addons/block_code/ui/blocks/entry_block/entry_block.gd#L22) doesn't go through the normal substitution path.

I guess a better way I hadn't thought of before is to use a dedicated class and then special case it within `TYPE_OBJECT` like `OptionData` is. E.g., `FunctionParameter` or something like that.